addons/mail/static/src/js/systray.js

```diff
@@ -138,14 +125,104 @@ var MessagingMenu = Widget.extend({
+        } else {
+            var channel = chat_manager.get_channel(channel_id);
+            if (channel) {
+                chat_manager.open_channel(channel);
+            }
+        }
+    },
+});
+
+/**
+ * Menu item appended in the systray part of the navbar, redirects to the next activities of all app
+ */
+var ActivityMenu = Widget.extend({
+    template:'mail.chat.ActivityMenu',
+    events: {
+        "click": "on_activity_menu_click",
+        "click .o_activity_filter_button, .o_mail_channel_preview": "on_activity_filter_click",
+    },
+    start: function () {
+        this.$activities_preview = this.$('.o_mail_navbar_dropdown_channels');
+        chat_manager.bus.on("activity_updated", this, this.update_counter);
+        chat_manager.is_ready.then(this.update_counter.bind(this));
+        this.update_activity_preview();
+        this.is_counter_updated = false;
+        return this._super();
+    },
+    is_open: function () {
+        return this.$el.hasClass('open');
+    },
+    update_counter: function (data) {
+        if (data) {
+            if (data.activity_created) {
+                this.activity_counter ++;
+            }
+            if (data.activity_deleted && this.activity_counter > 0) {
+                this.activity_counter --;
+            }
+            this.is_counter_updated = true;
+            this.$('.o_notification_counter').text(this.activity_counter);
+            this.$el.toggleClass('o_no_notification', !this.activity_counter);
         }
     },
+    get_activity_data: function(){
+        var self = this;
+
+        return self._rpc({
+            model: 'mail.activity',
+            method: 'get_activities',
+        }).then(function (data) {
+            self.is_counter_updated = false;
+            self.activities = data;
+            self.activity_counter = _.reduce(data, function(total_count, p_data){ return total_count + p_data.total_count; }, 0);
+            self.$('.o_notification_counter').text(self.activity_counter);
+            self.$el.toggleClass('o_no_notification', !self.activity_counter);
+        });
+    },
+    update_activity_preview: function () {
+        var self = this;
+        self.get_activity_data().then(function (){
+            self.$activities_preview.html(QWeb.render('mail.chat.ActivityMenuPreview', {
+                activities : self.activities
+            }));
+        });
+    },
+    on_activity_menu_click: function () {
+        if (!this.is_open() && this.is_counter_updated) {
+            this.update_activity_preview();
+        }
+    },
+    get_action_data: function (name, model, res_id) {
+        return {
+            type: 'ir.actions.act_window',
+            name: name,
+            res_model: model,
+            res_id: res_id,
+            domain: [['activity_user_id', '=', session.uid]],
+            views: res_id ? [[false, 'form'], [false, 'kanban']] : [[false, 'kanban'], [false, 'form']],
```

Comment:
Check if something like _get_form_view exists and use it. It makes 1 rpc call but we should land on the correct view